src/curl.cpp

```diff
@@ -3827,189 +3819,126 @@ bool S3fsMultiCurl::SetS3fsCurlObject(S3fsCurl* s3fscurl)
 
 int S3fsMultiCurl::MultiPerform(void)
 {
-  CURLMcode curlm_code;
-  int       still_running;
+  std::vector<pthread_t>   threads;
+  bool                     success = true;
+
+  for(s3fscurlmap_t::iterator iter = cMap_req.begin(); iter != cMap_req.end(); ++iter) {
+    pthread_t   thread;
+    S3fsCurl*   s3fscurl = (*iter).second;
+    int         rc;
+
+    rc = pthread_create(&thread, NULL, RequestPerformWrapper, static_cast<void*>(s3fscurl));
```

Comment:
This number equals the "parallel_count" argument (which defaults to 5).
Threads will be created only when handling files which are larger than "multipart_size" (defaults to 10MB).